memory-viz/src/memory_model.ts

```diff
@@ -1049,28 +1082,94 @@ export class MemoryModel {
     }
 
     /**
-     * Returns a copy of the input objects with width and height set properly.
+     * Returns a copy of the input objects with width, height, x and y coordinates set properly.
      *
      * @param objects - the list of objects (including stack-frames) to be drawn.
      */
     private setDimensionsAll(objects: DrawnEntity[]): DrawnEntityStrict[] {
-        const strict_objects: DrawnEntityStrict[] = [];
+        const objects_with_dimensions: DrawnEntityWithDimensions[] = [];
 
+        // Set width and height for all objects
         for (const object of objects) {
             // Get default width and height
             const default_dims = this.getDefaultDimensions(object);
 
             // Overwrite width and height if necessary
-            const strict_object = this.setDimensions(
+            const object_with_dimensions = this.setDimensions(
                 object,
                 default_dims.default_width,
                 default_dims.default_height
             );
 
-            strict_objects.push(strict_object);
+            objects_with_dimensions.push(object_with_dimensions);
+        }
+
+        const { stack_frames, other_items } = this.separateObjects(
+            objects_with_dimensions
+        );
+
+        // Set x and y coordinates for all stack frames
+        const { StackFrames, requiredHeight, requiredWidth } =
+            this.setStackFrameCoordinates(stack_frames);
+
+        // Determining the minimum width of the canvas.
```

Comment:
Somewhat similar to my last comment, I would move this section into the `this.setOtherItemsCoordinates` (and update `this.width` directly in there).